R/remote_path.R

```diff
@@ -1,36 +1,57 @@
-remote_report_pull_archive_path <- function(name, id, config, remote) {
-  src <- file.path(path_archive(remote), name, id)
-  dest <- file.path(path_archive(config$path), name, id)
-  copy_directory(src, dest, TRUE)
+##' @title Orderly remote at a different path
+##' @param path Path to the orderly store
+##' @param name Name of the remote
+##' @export
+orderly_remote_path <- function(path, name = NULL) {
+  R6_orderly_remote_path$new(path, name)
 }
 
 
-orderly_remote_path <- function(path) {
-  assert_file_exists(path)
-  path <- normalizePath(path, "/", mustWork = TRUE)
-  if (!file.exists(path_orderly_config_yml(path))) {
-    stop("Does not look like an orderly repository: ", squote(path))
-  }
-  structure(path,
-            class = c("orderly_remote_path",
-                      "orderly_remote_location"))
-}
-
-
-remote_report_names_path <- function(remote) {
-  orderly_list(orderly_config(remote))
-}
-
-
-remote_report_versions_path <- function(name, remote) {
-  config_remote <- orderly_config(remote)
-  d <- orderly_list_archive(config_remote)
-  d$id[d$name == name]
-}
-
-
-push_archive_path <- function(name, id, config, remote, overwrite = FALSE) {
-  src <- file.path(path_archive(config$path), name, id)
-  dest <- file.path(path_archive(remote), name, id)
-  copy_directory(src, dest, rollback_on_error = TRUE)
-}
+R6_orderly_remote_path <- R6::R6Class(
+  "orderly_remote_path",
+
+  public = list(
+    config = NULL,
+    name = NULL,
+
+    initialize = function(path, name) {
+      assert_file_exists(path)
+      path <- normalizePath(path, "/", mustWork = TRUE)
+      if (!file.exists(path_orderly_config_yml(path))) {
+        stop("Does not look like an orderly repository: ", squote(path))
+      }
+      self$config <- orderly_config(path)
+      self$name <- name %||% self$config$path
+      lockBinding(quote(config), self)
+      lockBinding(quote(name), self)
+    },
+
+    list = function() {
```

Comment:
Is `list_reports` clearer? Having functions with the same name as built in primitives makes me anxious.

Reply:
that would be less ambiguous with `list_versions`.

There's no danger of colliding with internal functions though (they're not keywords in R) because this is always called `thing$list()` which is equivalent to `thing[["list"]]()`